compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp

```diff
@@ -164,33 +164,54 @@ void SetSigProcMask(__sanitizer_sigset_t *set, __sanitizer_sigset_t *oldset) {
   CHECK_EQ(0, internal_sigprocmask(SIG_SETMASK, set, oldset));
 }
 
+#  if SANITIZER_LINUX
+// Deletes the specified signal from newset, if it is not present in oldset
+// Equivalently: newset[signum] = newset[signum] & oldset[signum]
+static void KeepUnblocked(__sanitizer_sigset_t &newset,
+                          __sanitizer_sigset_t &oldset, int signum) {
+  // FIXME: https://github.com/google/sanitizers/issues/1816
+  if (SANITIZER_ANDROID || !internal_sigismember(&oldset, signum))
+    internal_sigdelset(&newset, signum);
+}
+#  endif
+
 // Block asynchronous signals
 void BlockSignals(__sanitizer_sigset_t *oldset) {
-  __sanitizer_sigset_t set;
-  internal_sigfillset(&set);
+  __sanitizer_sigset_t currentset;
```

Comment:
unused current set warning is possible on non-linux.

Reply:
Non-Linux will still call `SetSigProcMask(&newset, oldset);` on line 214 (though it's effectively a no-op)

Reply:
Thanks, will fix

Reply:
we still need 
```
__sanitizer_sigset_t newset;
internal_sigfillset(&newset);

# if SANITIZER_LINUX
...
#endif

SetSigProcMask(&newset, oldset);
```

Reply:
Oops, thank you!